File: iosApp/Shared/Screens/ChatList/ChatListScreen.swift
//
// Created by Phil on 16.06.2021.
//

import SwiftUI
import SharedMobile

struct ChatListScreen: View {
    let state: ChatListFeature.State
    let listener: (ChatListFeature.Msg) -> Void

    var body: some View {
        NavigationBar(
            title: "Chat list"
        )
        ScrollView {
            LazyVStack {
                ForEachIndexed(state.listItems) { _, listItem in
                    Button {
                        listener(ChatListFeature.MsgSelectChat(userId: listItem.user.id))
                    } label: {
                        ChatListCell(item: listItem)
                    }
                }
            }
        }
        Spacer()
    }
}

File: modules/androidUi/src/main/kotlin/com/well/modules/androidUi/composableScreens/chatList/ChatListScreen.kt
package com.well.modules.androidUi.composableScreens.chatList

import com.well.modules.androidUi.customViews.Control
import com.well.modules.androidUi.customViews.NavigationBar
import com.well.modules.features.chatList.chatListFeature.ChatListFeature.Msg
import com.well.modules.features.chatList.chatListFeature.ChatListFeature.State
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.runtime.Composable

@Composable
fun ChatListScreen(
    state: State,
    listener: (Msg) -> Unit,
) {
    NavigationBar(
        title = "Chat list"
    )
    LazyColumn {
        items(state.listItems) { item ->
            Control(onClick = {
                listener(Msg.SelectChat(item.user.id))
            }) {
                ChatListCell(item = item)
            }
        }
    }
}